### PR TITLE
`browsingContext.reload`: add ignore cache E2E test

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #650

### CODE REVIEW

File: tests/test_browsing_context.py [hunk 1]
```diff
@@ -879,6 +879,104 @@ async def test_browsingContext_reload_waitComplete(websocket, context_id,
...
+            "redirectCount": 0,
+            "request": ANY_DICT,
+            "response": ANY_DICT,
+            "timestamp": ANY_TIMESTAMP,
+        },
+    }
+
+    # Assert command done.
+    response = await read_JSON_message(websocket)
+    assert response["result"] == {}
```
Comment (sadym-chromium) [inline tests/test_browsing_context.py:930]: NIT: check command `ID` as well please

File: tests/test_browsing_context.py [hunk 2]
```diff
@@ -879,6 +879,110 @@ async def test_browsingContext_reload_waitComplete(websocket, context_id,
...
+    url = "https://example.com/"
+
+    await subscribe(websocket, [
+        "network.beforeRequestSent",
+        "network.responseCompleted",
+    ])
+
+    await goto_url(websocket, context_id, url)
+
+    await send_JSON_command(
```
Comment (sadym-chromium) [inline tests/test_browsing_context.py:892]:
```suggestion
    command_id = await send_JSON_command(
```

File: tests/test_browsing_context.py [hunk 3]
```diff
@@ -879,6 +879,110 @@ async def test_browsingContext_reload_waitComplete(websocket, context_id,
...
+            "request": ANY_DICT,
+            "response": ANY_DICT,
+            "timestamp": ANY_TIMESTAMP,
+        },
+    }
+
+    # Assert command done.
+    response = await read_JSON_message(websocket)
+    assert response == {
+        "id": ANY_INT,
```
Comment (sadym-chromium) [inline tests/test_browsing_context.py:931]:
```suggestion
        "id": command_id,
```

**Comment by thiagowfx** [comment]
For now, both `true` and `false` for `ignoreCache` pass. We need something else to ensure the behavior is proper. @sadym-chromium any other ideas? Network events do not seem to be enough to differentiate them.

**Comment by sadym-chromium** [comment]
what about `fromCache` flag in the `network.responseCompleted` event?

**Comment by thiagowfx** [comment]
Good idea. Both tests have `fromCache: False` though.

File: tests/test_browsing_context.py [hunk 4]
```diff
@@ -871,6 +871,56 @@ async def test_browsingContext_reload_waitComplete(websocket, context_id,
...
+
+    response = await read_JSON_message(websocket)
+    assert response == {
+        "method": "network.responseCompleted",
+        "params": {
+            "context": context_id,
+            "navigation": ANY_STR,
+            "redirectCount": 0,
+            "request": ANY_DICT,
+            "response": AnyExtending({"fromCache": False}),
```
Comment (thiagowfx) [inline tests/test_browsing_context.py:916]: One would expect the correct value here is `not ignoreCache`, however it fails.

Comment (thiagowfx) [reply tests/test_browsing_context.py:916]: Upon discussing this in internal chat, it doesn't seem to be possible to detect this condition from `network.responseCompleted`.

**Pull request closed without merging**